gems/pending/postgres_ha_admin/failover_databases.rb

```diff
@@ -0,0 +1,78 @@
+require 'util/postgres_dsn_parser'
+require 'postgres_ha_admin/postgres_ha_logger'
+require 'pg'
+
+module PostgresHaAdmin
+  class FailoverDatabases
+    include PostgresHaAdmin
+
+    attr_reader :yml_file, :connection_hash, :log, :connection_hash
+
+    def initialize(config_dir, log_dir, connection_params_hash)
+      @yml_file = Pathname.new(config_dir).join(DB_YML_FILE)
+      @log = PostgresHaLogger.new(log_dir)
+      @connection_hash = connection_params_hash
+    end
+
+    def refresh_databases_list(connection = nil)
+      query_repmgr(connection)
+    end
+
+    def all_databases(connection = nil)
```

Comment:
If you passed in an object that responded to the same interface`all_databases`, the YamlDatabaseList(or a better name, not sure what `all_databases` means `all_standby_databases` or something else) class could  read from the yml, a MockYamlDatabaseList could be passed in with a predefined list of databases for testing.

My naming is horrible but doing something like that would allow you to remove this whole conditional by relying on objects that respond to the method to get the array of databases.